src/cluster.c

```diff
@@ -1354,15 +1354,17 @@ void addNodeReplyForClusterSlot(client *c, clusterNode *node, int start_slot, in
 
 void clearCachedClusterSlotsResponse(void) {
     for (connTypeForCaching conn_type = CACHE_CONN_TCP; conn_type < CACHE_CONN_TYPE_MAX; conn_type++) {
-        if (server.cached_cluster_slot_info[conn_type]) {
-            sdsfree(server.cached_cluster_slot_info[conn_type]);
-            server.cached_cluster_slot_info[conn_type] = NULL;
+        for (int resp = 0; resp <= 3; resp++) {
```

Comment:
minor thing, since we don't have 0 and 1, we can start from 2
```suggestion
        for (int resp = 2; resp <= 3; resp++) {
```

Reply:
how about using enum for RESP versions?

Reply:
I would be OK moving to an enum, but not in this PR. The resp value is used in several other places throughout the codebase and it would add a bunch more unnecessary touchpoints.

Reply:
I tend to fix this issue first and then implement RESP enum in a new PR, WDYT?